aten/src/ATen/autocast_mode.cpp

```diff
@@ -507,18 +507,15 @@ TORCH_LIBRARY_IMPL(aten, AutocastCPU, m) {
   KERNEL_CPU(matmul, lower_precision_fp)
   KERNEL_CPU(conv_tbc, lower_precision_fp)
   KERNEL_CPU(mkldnn_rnn_layer, lower_precision_fp)
+  KERNEL_CPU(conv_transpose1d, lower_precision_fp)
+  KERNEL_CPU2(conv_transpose2d, input, lower_precision_fp)
+  KERNEL_CPU2(conv_transpose3d, input, lower_precision_fp)
 
   // fp32 cast policy
-  KERNEL_CPU(conv_transpose1d, fp32)
-  KERNEL_CPU2(conv_transpose2d, input, fp32)
-  KERNEL_CPU2(conv_transpose3d, input, fp32)
   KERNEL_CPU(avg_pool3d, fp32)
   KERNEL_CPU(binary_cross_entropy, fp32)
   KERNEL_CPU(grid_sampler, fp32)
   KERNEL_CPU(polar, fp32)
-  KERNEL_CPU(prod, fp32)
-  KERNEL_CPU2(prod, dim_int, fp32)
-  KERNEL_CPU2(prod, dim_Dimname, fp32)
```

Comment:
Why is this one needed?

Reply:
prod should be kept in the fp32 cast policy since we found that tensoriterator reduction related ops may have precision issue for bf16. We will work on it in the future. Now I will keep prod.